polaris-service/build.gradle.kts

```
@@ -0,0 +1,230 @@
+      "UpdateNamespacePropertiesRequest" to
+        "org.apache.iceberg.rest.requests.UpdateNamespacePropertiesRequest",
+      "UpdateNamespacePropertiesResponse" to
+        "org.apache.iceberg.rest.responses.UpdateNamespacePropertiesResponse",
+      "CommitTransactionRequest" to "org.apache.iceberg.rest.requests.CommitTransactionRequest",
+      "CreateViewRequest" to "org.apache.iceberg.rest.requests.CreateViewRequest",
+      "RegisterTableRequest" to "org.apache.iceberg.rest.requests.RegisterTableRequest",
+      "IcebergErrorResponse" to "org.apache.iceberg.rest.responses.ErrorResponse",
+      "OAuthError" to "org.apache.iceberg.rest.responses.ErrorResponse",
+
+      // Custom types defined below
+      "CommitViewRequest" to "io.polaris.service.types.CommitViewRequest",
+      "TokenType" to "io.polaris.service.types.TokenType",
+      "CommitTableRequest" to "io.polaris.service.types.CommitTableRequest",
+      "NotificationRequest" to "io.polaris.service.types.NotificationRequest",
+      "TableUpdateNotification" to "io.polaris.service.types.TableUpdateNotification",
+      "NotificationType" to "io.polaris.service.types.NotificationType"
+    )
+}
+
+tasks.register<GenerateTask>("generatePolarisService").configure {
+  inputSpec = "$rootDir/spec/polaris-management-service.yml"
+  generatorName = "jaxrs-resteasy"
+  outputDir = "$projectDir/build/generated"
+  apiPackage = "io.polaris.service.admin.api"
+  modelPackage = "io.polaris.core.admin.model"
+  ignoreFileOverride = "$rootDir/.openapi-generator-ignore"
+  removeOperationIdPrefix = true
+  templateDir = "$rootDir/server-templates"
+  globalProperties.put("apis", "")
+  globalProperties.put("models", "false")
+  globalProperties.put("apiDocs", "false")
+  globalProperties.put("modelTests", "false")
+  configOptions.put("useBeanValidation", "true")
+  configOptions.put("sourceFolder", "src/main/java")
+  configOptions.put("useJakartaEe", "true")
+  configOptions.put("generateBuilders", "true")
+  configOptions.put("generateConstructorWithAllArgs", "true")
+  additionalProperties.put("apiNamePrefix", "Polaris")
+  additionalProperties.put("apiNameSuffix", "Api")
+  additionalProperties.put("metricsPrefix", "polaris")
+  serverVariables.put("basePath", "api/v1")
+}
+
+tasks.named("compileJava").configure { dependsOn("openApiGenerate", "generatePolarisService") }
+
+sourceSets {
+  main { java { srcDir(project.layout.buildDirectory.dir("generated/src/main/java")) } }
+}
+
+tasks.named<Test>("test").configure {
+  if (System.getenv("AWS_REGION") == null) {
+    environment("AWS_REGION", "us-west-2")
+  }
+  jvmArgs("--add-exports", "java.base/sun.nio.ch=ALL-UNNAMED")
+  useJUnitPlatform()
+  maxParallelForks = 4
+}
+
+tasks.register<JavaExec>("runApp").configure {
+  if (System.getenv("AWS_REGION") == null) {
+    environment("AWS_REGION", "us-west-2")
+  }
+  classpath = configurations["runtimeClasspath"]
```

This isn't working properly on my machine. The below change makes it work:

```suggestion
  classpath = sourceSets["main"].runtimeClasspath
```

Otherwise I get:

```
> Task :polaris-service:runApp FAILED
Error: Could not find or load main class io.polaris.service.PolarisApplication
Caused by: java.lang.ClassNotFoundException: io.polaris.service.PolarisApplication
```

---

thanks, fixed